src/lib.rs

```diff
@@ -327,4 +338,30 @@ mod tests {
         assert_eq!(c.take_or_clone_val(), Some(3));
         assert_eq!(c1.take_or_clone_val(), Some(2));
     }
+
+    #[test]
+    fn test_hash() {
+        fn calculate_hash<T: Hash>(t: &T) -> u64 {
+            let mut s = DefaultHasher::new();
+            t.hash(&mut s);
+            s.finish()
+        }
+
+        let c1 = Cactus::new().child(4).child(3);
+        let c2 = Cactus::new().child(4).child(3);
+        assert_eq!(calculate_hash(&c1), calculate_hash(&c2));
+        // The next test is fragile in theory although probably not in practise. Since there's no
+        // guarantee that two distinct things will map to distinct hashes, it's perfectly possible
+        // that a hasher returns the same value for two distinct cactuses. But this isn't hugely
```

Comment:
If I were feeling pedantic: cactuses -> cacti :)

[According to wikipedia]

Reply:
I think cactuses is OK (I did use "cactii" (with two "I"s?!) in the PR message for amusement). http://grammarist.com/usage/cacti-cactuses/

Reply:
Fine, it doesn't really matter TBH :)